Firmware/Marlin_main.cpp

```diff
@@ -235,6 +235,7 @@ uint8_t newFanSpeed = 0;
   #endif
 
 static bool cancel_heatup = false;
+bool CooldownNoWait;
```

Comment:
After re-reading this, it seems that wait_for_heater is also called from restore_extruder_temperature_from_ram, so the last value of M109 and M190 will affect the behavior of wait_for_heater, which is incorrect.

I don't see a reason why we need a global for this. I would add a parameter to wait_for_heater, with the default being false to match the previous behavior.